kuma/users/views.py

```diff
@@ -80,10 +82,84 @@ def ban_user(request, user_id):
     return render(request,
                   'users/ban_user.html',
                   {'form': form,
-                   'user_to_ban': user,
+                   'detail_user': user,
                    'common_reasons': common_reasons})
 
 
+@permission_required('users.add_userban')
+def ban_user_and_cleanup(request, user_id):
+    """
+    A page to ban a user for the reason of "Spam" and mark the user's revisions
+    and page creations as spam, reverting as many of them as possible.
+    """
+    try:
+        user = User.objects.get(pk=user_id)
+    except User.DoesNotExist:
+        raise Http404
+
+    # Is this user already banned?
+    user_ban = UserBan.objects.filter(user=user, is_active=True)
+
+    # Get revisions for the past 3 days for this user
+    date_three_days_ago = datetime.now().date() - timedelta(days=3)
+    revisions = user.created_revisions.prefetch_related('document')
+    revisions = revisions.defer('content', 'summary').order_by('-created')
```

Comment:
``.order_by('-id')`` gives the same results as ``.order_by('-created')``, but can be 100x faster in some ugly cases, as I'm learning in PR #3890